pandas/_libs/algos_rank_helper.pxi.in

```diff
@@ -226,12 +226,10 @@ def rank_2d_{{dtype}}(object in_arr, axis=0, ties_method='average',
 
     keep_na = na_option == 'keep'
 
-    in_arr = np.asarray(in_arr)
-
     if axis == 0:
-        values = in_arr.T.copy()
+        values = np.asarray(in_arr).T.copy()
     else:
-        values = in_arr.copy()
+        values = np.asarray(in_arr).copy()
```

Comment:
Why do we need this change?

Reply:
Because without it cython raises at compile-time because it in_arr and values have different types